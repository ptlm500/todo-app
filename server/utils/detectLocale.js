const acceptLanguage = require('accept-language');
const config = require('../config.json');

function detectLocale(req) {
  acceptLanguage.languages(config.ACCEPTED_LANGUAGES);

  const cookieLocale = req.cookies.locale;

  return acceptLanguage.get(cookieLocale || req.headers['accept-language']) ||
    config.DEFAULT_LOCALE;
}

module.exports = detectLocale;
