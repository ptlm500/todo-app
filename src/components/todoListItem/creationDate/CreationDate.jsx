import React from 'react';
import PropTypes from 'prop-types';
import { FormattedDate } from 'react-intl';
import styled from 'styled-components';
import { theme } from '../../../styles';

function CreationDate(props) {
  const {
    className,
    creationDate
  } = props;

  return (
    <a className={className}>
      <FormattedDate
        value={new Date(parseInt(creationDate))}
        year="numeric"
        month="long"
        day="2-digit"
        hour="numeric"
        minute="numeric"
      />
    </a>
  );
}

CreationDate.propTypes = {
  className: PropTypes.string.isRequired,
  creationDate: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
};

export default styled(CreationDate)`
  ${theme.font.bodySize};
  color: ${theme.colour.secondaryText}
`;
