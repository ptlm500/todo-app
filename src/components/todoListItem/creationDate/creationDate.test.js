import React from 'react';
import { renderWithIntl } from '../../../../test/utils';
import CreationDate from './CreationDate';
import 'jest-styled-components';

describe('CreationDate', () => {
  it('renders the creation date using FormattedMessage', () => {
    const tree = renderWithIntl(
      <CreationDate className='test-class' creationDate='1582029687000' />
    );

    expect(tree).toMatchSnapshot();
  });
});
