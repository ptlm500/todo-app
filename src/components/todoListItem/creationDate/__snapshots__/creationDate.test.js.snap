// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreationDate renders the creation date using FormattedMessage 1`] = `
<a
  className="sc-AxjAm ELMTo test-class"
>
  February 18, 2020, 12:41 PM
</a>
`;
